### ISSUE TITLE
Royale Not Reading asconfig at compilation

### ISSUE THREAD

**Issue by angelhdz** [BowlerHatLLC/vscode-as3mxml#326]
Hello again Josh! 

Sorry to bother you, but the guys at Apache Royale are ignoring my messages everywhere...

I've read the documentation many times. I can compile my Royale app well. The issue is using the Jewel Themes. 

At first, I didn't know how to add the Jewel Themes. Months trying. Until I found a post in Google showing to set the "theme" attribute in the asconfig file.

I did it and still didn't get the themes, using either relative or full paths to the themes.

So I was about to give up, and every time I hit Run Build Task, I got an annoying Chrome browser pop up window to debug, even if I had the debug attribute set to false. That made me think that the Royale compiler isn't loading the asconfig file, but the nextgenas extension language server is loading it at authoring time and for linting.

So, if every time I run build task it is still showing debug: "true", that means the Royale compiler is not loading the asconfig file, and hence why the Jewel themes aren't being applied either.

Is this a problem with your extension? Is a Royale compiler issue? I will try on Moonshine IDE in the meantime.

Thank you for the information you can bring. I'm very hyped to try Royale components to build my apps, but at the same time, due to issues like this (and other issues I had configuring Royale), I am like losing interest.

Blessings and have a nice week!

**Comment by joshtynjala** [comment]
The compilers know nothing about asconfig.json. It is a file format that I created for the VSCode extension. I also created asconfigc, a tool that reads this file format and runs the compiler with the appropriate options. When you run one of the built-in compile tasks in VSCode, it runs asconfigc.

I'm not aware of any issues compiling with a Jewel theme in VSCode. Carlos Rovira, the creator of the Jewel components, uses VSCode. Although, he builds with Maven, so there is a slight difference in his workflow.

Anyway, you can see a theme being used in the TourDeJewel example here:

https://github.com/apache/royale-asjs/blob/develop/examples/royale/TourDeJewel/asconfig.json

Is that how you've specified the theme in your asconfig.json file?

**Comment by joshtynjala** [comment]
However, I see that the themes are supposed to be built as SWCs, so it's possible that you're actually supposed to reference the *.swc* file in the theme compiler option, and not a *.css* file. If you haven't tried referencing the theme *.swc* file, you should try that.

**Comment by angelhdz** [comment]
Tried:

`"theme": "${royalelib}/themes/JewelTheme/src/main/resources/defaults.css",`

And:

`"theme": "${royalelib}/themes/JewelTheme/target/JewelTheme.swc",`

None work.

Also, as you can see in the picture, I created the build.xml and pom.xml files, exactly from one of the Jewel Examples and added them in the same directory with the asconfig.json. 

I don't know what else to do.

![2019-02-21_08-44-27](https://user-images.githubusercontent.com/10191085/53169996-4ab15880-35b5-11e9-92bb-e6513d329548.png)
![code_2019-02-21_08-43-41](https://user-images.githubusercontent.com/10191085/53169997-4b49ef00-35b5-11e9-84f8-9465c2a91016.png)
![2019-02-21_08-45-14](https://user-images.githubusercontent.com/10191085/53169998-4b49ef00-35b5-11e9-82a8-27fb316b3f1d.png)

**Comment by angelhdz** [comment]
### SOLVED

Hey Josh!

I finally got my issue solved!

Turned out  that the Jewel themes were applied, but I need to use

`<j:beads>
        <js:ContainerDataBinding/>
    </j:beads>`

in order to get the colors working.


### QUESTION

I don't know if I need to open a new Issue here, but I have a question regarding nextgenactionscript extension:

To get updated output, I need to first  **Run Build Task** and then I can hit CNTRL + ENTER to debug.

Is there a way of doing that in one step?

Thank you! :)

**Comment by angelhdz** [comment]
Nevermind. I think is a bug of the CNTRL + ENTER (Build and Debug) command. I will open a new Issue.

**Comment by joshtynjala** [comment]
Ctrl+Enter is supposed to run the equivalent of the built-in task "ActionScript: compile debug" before starting the debugger. If the build fails, the debugger should not start. I use Ctrl+Enter all the time, so you may have found an interesting edge case.
